keanu-project/src/main/java/io/improbable/keanu/distributions/continuous/Beta.java

```diff
@@ -7,66 +7,99 @@
 import io.improbable.keanu.distributions.ContinuousDistribution;
 import io.improbable.keanu.distributions.hyperparam.Diffs;
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.LogProbGraph;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.PlaceHolderDoubleVertex;
 
 public class Beta implements ContinuousDistribution {
 
-    private final DoubleTensor alpha;
-    private final DoubleTensor beta;
-    private final DoubleTensor xMin;
-    private final DoubleTensor xMax;
+    private final DoubleVertex x;
+    private final DoubleVertex alpha;
+    private final DoubleVertex beta;
+    private final DoubleVertex xMin;
+    private final DoubleVertex xMax;
 
-    public static ContinuousDistribution withParameters(DoubleTensor alpha, DoubleTensor beta, DoubleTensor xMin, DoubleTensor xMax) {
-        return new Beta(alpha, beta, xMin, xMax);
+    private final LogProbGraph logProbGraph;
+
+    public static Beta withParameters(DoubleVertex x, DoubleVertex alpha, DoubleVertex beta, DoubleVertex xMin, DoubleVertex xMax) {
+        return new Beta(x, alpha, beta, xMin, xMax);
```

Comment:
The value of `x` is never used. Its only purpose is to act as a key in your `LogProbGraph` map. I think therefore it doesn't need to be a constructor argument - we can keep the constructor signature unchanged.